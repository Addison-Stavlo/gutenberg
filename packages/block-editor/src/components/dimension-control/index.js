/**
 * WordPress dependencies
 */
import {
	BaseControl,
	IconButton,
	Icon,
} from '@wordpress/components';
import { __, sprintf } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import sizesTable, { findSizeBySlug } from './sizes';
import DimensionButtons from './buttons';

function DimensionControl( props ) {
	const { title, property, device = 'all', deviceIcon = 'desktop', id, currentSize, onSpacingChange, onReset } = props;

	/**
	 * Determines the size from the size slug (eg: `medium`)
	 * and decides whether to call the change or reset callback
	 * handlers
	 * @param  {Object} event the click event for size buttons
	 * @return {void}
	 */
	const onChangeSpacingSize = ( event ) => {
		const theSize = findSizeBySlug( sizesTable, event.target.value );

		if ( ! theSize ) {
			return;
		}

		if ( currentSize === theSize.slug ) {
			resetSpacing();
		} else {
			onSpacingChange( theSize.slug );
		}
	};

	/**
	 * Applies the callback to handle resetting
	 * a dimension spacing values
	 * @return {void}
	 */
	const resetSpacing = () => onReset();

	return (
		<BaseControl
			id={ `block-spacing-${ property }-desktop-${ id }` }
			help={ sprintf( __( 'Select the %s for this Block' ), property ) }
			className="block-editor-dimension-control"
		>
			<div className="block-editor-dimension-control__header">
				<BaseControl.VisualLabel className="block-editor-dimension-control__header-label">
					<Icon
						icon={ deviceIcon || device }
						label={ device }
					/>
					{ sprintf( __( '%s (%s devices)' ), title, device ) }
				</BaseControl.VisualLabel>

				<IconButton
					icon="controls-repeat"
					label={ sprintf( __( 'Reset %s' ), title ) }
					isDefault
					isSmall
					onClick={ resetSpacing }
				/>
			</div>

			<DimensionButtons
				{ ...props }
				device={ device }
				currentSize={ currentSize }
				onChangeSpacingSize={ onChangeSpacingSize }
				sizes={ sizesTable }
			/>

		</BaseControl>
	);
}

export default DimensionControl;
