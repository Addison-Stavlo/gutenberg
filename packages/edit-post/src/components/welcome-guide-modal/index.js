/**
 * WordPress dependencies
 */
import { useSelect, useDispatch } from '@wordpress/data';
import { Modal } from '@wordpress/components';
import { __, sprintf } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import Guide from './guide';
import { CanvasImage, EditorImage, BlockLibraryImage, inserterIconHTML } from './vectors';

export default function WelcomeGuideModal() {
	const isOpen = useSelect( ( select ) =>
		select( 'core/edit-post' ).isFeatureActive( 'welcomeGuide' ) );

	const { toggleFeature } = useDispatch( 'core/edit-post' );

	if ( ! isOpen ) {
		return null;
	}

	const closeModal = () => {
		toggleFeature( 'welcomeGuide' );
	};

	return (
		<Modal className="edit-post-welcome-guide-modal" onRequestClose={ closeModal }>
			<Guide onRequestClose={ closeModal }>
				<Guide.Page>
					<CanvasImage />
					<div className="edit-post-welcome-guide-modal__text">
						<h1>{ __( 'Welcome to the block editor' ) }</h1>
						<p>
							{ __( 'In the WordPress editor, each paragraph, image, or video is presented as a distinct “block” of content.' ) }
						</p>
					</div>
				</Guide.Page>
				<Guide.Page>
					<EditorImage />
					<div className="edit-post-welcome-guide-modal__text">
						<h1>{ __( 'Make each block your own' ) }</h1>
						<p>
							{ __( 'Each block comes with its own set of controls for changing things like color, width, and alignment. These will show and hide automatically when you have a block selected.' ) }
						</p>
					</div>
				</Guide.Page>
				<Guide.Page>
					<BlockLibraryImage />
					<div className="edit-post-welcome-guide-modal__text">
						<h1>{ __( 'Get to know the block library' ) }</h1>
						<p dangerouslySetInnerHTML={ {
							__html: sprintf(
								/* translators: %s: HTML which displays the inserter icon. */
								__( 'All of the blocks available to you live in the Block Library. You’ll find it wherever you see the %s icon.' ),
								inserterIconHTML
							),
						} } />
					</div>
				</Guide.Page>
			</Guide>
		</Modal>
	);
}
