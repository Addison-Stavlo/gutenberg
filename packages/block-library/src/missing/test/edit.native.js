/**
 * External dependencies
 */
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
jest.mock( 'Platform', () => {
	const Platform = require.requireActual( 'Platform' );
	Platform.OS = 'ios';
	return Platform;
} );

/**
 * WordPress dependencies
 */
import { BottomSheet, Icon } from '@wordpress/components';
jest.mock( '@wordpress/blocks' );
jest.mock( '../styles.scss', () => {
	return {
		infoIcon: { size: 32 },
		unsupportedBlockIcon: { color: 'white' },
	};
} );

/**
 * Internal dependencies
 */
import UnsupportedBlockEdit from '../edit.native.js';

const defaultAttributes = {
	originalName: 'missing/block/title',
};

const getTestComponentWithContent = ( attributes = defaultAttributes ) => {
	return renderer.create( <UnsupportedBlockEdit attributes={ attributes } /> );
};

describe( 'Missing block', () => {
	it( 'renders without crashing', () => {
		const component = getTestComponentWithContent();
		const rendered = component.toJSON();
		expect( rendered ).toMatchSnapshot();
	} );

	describe( 'help modal', () => {
		it( 'renders help icon', () => {
			const component = getTestComponentWithContent();
			const testInstance = component.root;
			const icons = testInstance.findAllByType( Icon );
			expect( icons.length ).toBe( 2 );
			expect( icons[ 0 ].props.icon ).toBe( 'editor-help' );
		} );

		it( 'renders info icon on modal', () => {
			const component = getTestComponentWithContent();
			const testInstance = component.root;
			const bottomSheet = testInstance.findByType( BottomSheet );
			const children = bottomSheet.props.children.props.children;
			expect( children.length ).toBe( 4 ); // 4 children in the bottom sheet: the icon, the "isn't yet supported" title, the "We are working hard..." message and the "Close" button
			expect( children[ 0 ].props.icon ).toBe( 'editor-help' );
		} );

		it( 'renders unsupported text on modal', () => {
			const component = getTestComponentWithContent();
			const testInstance = component.root;
			const bottomSheet = testInstance.findByType( BottomSheet );
			const children = bottomSheet.props.children.props.children;
			expect( children[ 1 ].props.children ).toBe( defaultAttributes.originalName + ' isn\'t yet supported on WordPress for iOS' );
		} );

		it( 'renders close button on modal', () => {
			const component = getTestComponentWithContent();
			const testInstance = component.root;
			const bottomSheet = testInstance.findByType( BottomSheet );
			const children = bottomSheet.props.children.props.children;
			expect( children[ 3 ].props.children.props.children ).toBe( 'Close' );
		} );
	} );

	it( 'renders admin plugins icon', () => {
		const component = getTestComponentWithContent();
		const testInstance = component.root;
		const icons = testInstance.findAllByType( Icon );
		expect( icons.length ).toBe( 2 );
		expect( icons[ 1 ].props.icon ).toBe( 'admin-plugins' );
	} );

	it( 'renders title text without crashing', () => {
		const component = getTestComponentWithContent();
		const testInstance = component.root;
		const texts = testInstance.findAllByType( Text );
		expect( texts.length ).toBe( 2 );
		expect( texts[ 0 ].props.children ).toBe( 'missing/block/title' );
		expect( texts[ 1 ].props.children ).toBe( 'Unsupported' );
	} );
} );
