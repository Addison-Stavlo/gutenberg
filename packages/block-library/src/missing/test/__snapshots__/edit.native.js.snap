// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missing block renders without crashing 1`] = `
<View>
  <View>
    <View
      accessibilityLabel="Help icon"
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 3,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 1,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          Svg
        </View>
      </View>
    </View>
  </View>
  Svg
  <Text>
    missing/block/title
  </Text>
  Modal
</View>
`;
